app/controllers/provider/sessions_controller.rb

```diff
@@ -87,12 +88,13 @@ def auth_params
   end
 
   def sso_params
-    params.permit(:token, :expires_at, :redirect_url, :system_name, :code).merge(request: request)
+    params.permit(%i[token expires_at redirect_url system_name code]).merge(request: request).to_h
   end
 
   def session_return_to
-    if params[:return_to]
-      return_to = safe_return_to(params[:return_to])
+    return_to_params = params.require(:return_to)
```

Comment:
params.require will raise an error if empty. It is unlikely that you want that behaviour